python_modules/libraries/dagster-dbt/dagster_dbt/rpc/resources.py

```diff
@@ -143,7 +144,7 @@ def status(self):
             Response: the HTTP response from the dbt RPC server.
         """
         data = self._default_request(method="status")
-        return self._post(data=json.dumps(data))
+        return DbtRpcClient._get_result(self, data=json.dumps(data))
```

Comment:
I think it would make sense to keep the _post() function around , and then define _get_result() for DbtRpcClient in in terms of this _post() function.

Then, poll, ps, etc. can all just call self._post() without an issue, as DbtRpcSyncClient will also inherit the _post capability.

Reply:
👍